src/pages/fi-gen-open-bal/form/GenerateOpening.js

```diff
@@ -0,0 +1,108 @@
+import { Grid } from '@mui/material'
+import * as yup from 'yup'
+import { useContext } from 'react'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useForm } from 'src/hooks/form'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+
+import CustomDatePicker from 'src/components/Inputs/CustomDatePicker'
+import { RGFinancialRepository } from 'src/repositories/RGFinancialRepository'
+import { formatDateToApi } from 'src/lib/date-helper'
+
+export default function GenerateOpening({ _labels, access }) {
+  const { postRequest } = useContext(RequestsContext)
+
+  const { formik } = useForm({
+    initialValues: { fiscalYear: null, tbFiscalYear: null, recordId: 'N/A', tbendDate: null },
+    enableReinitialize: true,
+    maxAccess: access,
+    validateOnChange: true,
+
+    validationSchema: yup.object({
+      fiscalYear: yup.string().required(' '),
+      tbendDate: yup.string().required(' '),
+      tbFiscalYear: yup.string().required(' ')
+    }),
+    onSubmit: async obj => {
+      try {
+        const { recordId, ...rest } = obj
+
+        const response = await postRequest({
+          extension: RGFinancialRepository.FiOpeningBalance.gen,
+          record: JSON.stringify({ ...obj, tbendDate: formatDateToApi(obj.tbendDate) })
+        })
+
+        toast.success('Record Success')
+
+        invalidate()
```

Comment:
where is this declared? I can't see no invalidate. This errors and stays hidden.